crates/common/src/operation.rs

```diff
@@ -121,23 +154,50 @@ pub struct RegisterServiceArgs {
 
 #[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
 pub enum ServiceChallenge {
-    Signed(PublicKey),
+    Signed(VerifyingKey),
 }
 
 impl From<SigningKey> for ServiceChallenge {
     fn from(sk: SigningKey) -> Self {
-        ServiceChallenge::Signed(PublicKey::Ed25519(sk.verifying_key().as_bytes().to_vec()))
+        ServiceChallenge::Signed(sk.verifying_key())
     }
 }
 
 #[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
 /// Common structure for operations involving keys (adding or revoking).
 pub struct KeyOperationArgs {
     pub id: String,                 // Account ID
-    pub value: PublicKey,           // Public key being added or revoked
+    pub value: VerifyingKey,        // Public key being added or revoked
     pub signature: SignatureBundle, // Signature to authorize the action
 }
 
+#[derive(Clone)]
+pub enum SigningKey {
```

Comment:
In the next PR, SigningKey and VerifyingKey should be extracted into its own file in the crate. These files are doing too much. But Jonas is also working on these files atm so it should be done in this PR